### DISCUSSION TITLE
Update / Reload upon reveal of the NvimTree

### DISCUSSION THREAD

**Discussion by juhaku** [nvim-tree/nvim-tree.lua#2517] [Q&A]
Hey,

I'd like a support for functionality where the diagnostics status and what not would be updated / reloaded upon NvimTree  reveal? Is that currently possible? I did not see any configuration option that would enable such thing.

EDIT: There is `nvim-tree.auto_reload_on_write` that allows tree to update while being opened and a file is being written to. Also there is `tree.reload()` API that is tied to `R` command command and `reload_on_bufenter` configuration flag but non of these seem to give remedy to the situation. Like hitting `R` button does not seem to affect much to the state of the tree at least the diagnostics does not get updated. Same happens no matter I travel between different buffers. The only way for diagnostics to actually get updated seems to be by way of writing on buffer while keeping the NvimTree open. 

-Juha

**Comment by alex-courtis** [comment]
> I'd like a support for functionality where the diagnostics status and what not would be updated / reloaded upon NvimTree reveal? 

I'm not quite sure what you are looking for.

Do you want to filter some statuses? Is there an issue with some statuses not being shown?

**Reply by juhaku** [reply]
Here's a picture for a reference.

The thing is, when the `NvimTree` is open and a buffer is being written to the diagnostic icons and statuses gets updated to the `NvimTree` in realtime. However if I close the `NvimTree` and update the buffer with some content e.g. I would correct some errors that otherwise vim lsp diagnostics would catch and then I open the `NvimTree` again the old status is being shown in the `NvimTree`. It would be fantastic if the diagnostic information on that tree would be refreshed when `NvimTree` is revealed or reopened. Then it would help to catch issues earlier if some of the files has some diagnostic errors. 

Does this explanation make it any clearer? :slightly_smiling_face: 

![image](https://github.com/nvim-tree/nvim-tree.lua/assets/26358664/f1eac2d2-1eb5-4179-8823-fecbcd82ceb3)

**Reply by alex-courtis** [reply]
Thank you. Raised #2519
